src/lib/gundb/UserStorage.js

```diff
@@ -0,0 +1,44 @@
+//@flow
+import { default as goodWallet, GoodWallet } from '../wallet/GoodWallet'
+import logger from '../logger/pino-logger'
+import initGunDB from './gundb'
+
+export type GunDBUser = {
+  alias: string,
+  epub: string,
+  pub: string
+}
+class UserStorage {
+  wallet: GoodWallet
+  profile: any
+  user: GunDBUser
+
+  constructor() {
+    this.wallet = goodWallet
+    this.wallet.ready.then(() => this.init())
```

Comment:
Having init in the constructor makes hard to know if the user is loaded. How should we use userStorage?

Reply:
App shouldn't work if wallet isn't ready.
add this ticket #163847881